mico-core/src/main/java/io/github/ust/mico/core/service/MicoStatusService.java

```diff
@@ -79,46 +120,59 @@ public MicoApplicationStatusDTO getApplicationStatus(MicoApplication micoApplica
                 micoService.getShortName(), micoService.getVersion(), e.getMessage());
         }
-        if (!deploymentOptional.isPresent()) {
+        if (deploymentOptional.isPresent()) {
+            Deployment deployment = deploymentOptional.get();
+            serviceStatus.setAvailableReplicas(deployment.getStatus().getAvailableReplicas());
+            serviceStatus.setRequestedReplicas(deployment.getSpec().getReplicas());
+        } else {
             log.warn("There is no deployment of the MicoService '{}' '{}'. Continue with next one.",
                 micoService.getShortName(), micoService.getVersion());
+            return serviceStatus.setErrorMessages(CollectionUtils.listOf("No deployment of " + micoService.getShortName() + " " + micoService.getVersion() + " is available."));
         }
-
-        Deployment deployment = deploymentOptional.get();
-        MicoServiceStatusDTO serviceStatus = new MicoServiceStatusDTO();
         serviceStatus.setName(micoService.getName());
         serviceStatus.setShortName(micoService.getShortName());
         serviceStatus.setVersion(micoService.getVersion());
-        serviceStatus.setAvailableReplicas(deployment.getStatus().getAvailableReplicas());
-        serviceStatus.setRequestedReplicas(deployment.getSpec().getReplicas());
 
-        // Get status information for service interfaces of a serivce
+        // Get status information for service interfaces of a service
         List<MicoServiceInterfaceDTO> micoServiceInterfaceDTOList = getServiceInterfaceStatus(micoService);
         serviceStatus.setInterfacesInformation(micoServiceInterfaceDTOList);
 
+        // Return the names of all applications that are using this service
+        List<MicoApplication> usingApplications = micoApplicationRepository.findAllByUsedService(micoService.getShortName(), micoService.getVersion());
+        for (MicoApplication micoApplication: usingApplications) {
+            MicoUsingApplicationDTO usingApplication = new MicoUsingApplicationDTO(micoApplication.getName(), micoApplication.getShortName(), micoApplication.getVersion());
+            serviceStatus.getUsingApplications().add(usingApplication);
+        }
+
         // Get status information for all pods of a service
+        int averageCpuLoad = 0;
+        int averageMemoryUsage = 0;
         List<Pod> podList = micoKubernetesClient.getPodsCreatedByDeploymentOfMicoService(micoService);
         List<KubernetesPodInfoDTO> podInfos = new LinkedList<>();
         for (Pod pod : podList) {
             KubernetesPodInfoDTO podInfo = getUiPodInfo(pod);
+            averageCpuLoad += podInfo.getMetrics().getCpuLoad();
+            averageMemoryUsage += podInfo.getMetrics().getMemoryUsage();
             podInfos.add(podInfo);
         }
-        serviceStatus.setPodInfo(podInfos);
+        serviceStatus.setAverageCpuLoad(averageCpuLoad/podList.size());
```

Comment:
Please use whitespaces before AND after operators or consider using an IDE auto format.